src/app/shared/countdown-button.component.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ChangeDetectionStrategy, Component, EventEmitter, Injector, Input, OnInit, Output
+} from '@angular/core';
+import { AbstractComponent } from 'app/shared/abstract.component';
+import { LayoutService } from 'app/shared/layout.service';
+import { BehaviorSubject, Observable, Subscription, timer } from 'rxjs';
+
+/**
+ * A button that displays a countdown after be clicked.
+ */
+@Component({
+  selector: 'countdown-button',
+  templateUrl: 'countdown-button.component.html',
+  styleUrls: ['countdown-button.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CountdownButtonComponent extends AbstractComponent implements OnInit {
+
+  text$ = new BehaviorSubject<string>(null);
+  disabled$ = new BehaviorSubject<boolean>(false);
+
+  @Input() first: boolean;
+  @Input() buttonText: string;
```

Comment:
I'd suggest to rename property to `label`.
As a general pattern, a class Person shouldn't have an attribute personName.
Similarly, don't use button... in attributes of a button.